sway-lib-std/src/tx.sw

```diff
@@ -10,41 +10,95 @@ use ::mem::read;
 use ::option::Option;
 
 ////////////////////////////////////////
-// Transaction fields
+// GTF Immediates
 ////////////////////////////////////////
 
-// The transaction starts at
-// TX_START = 32 + MAX_INPUTS*(32+8) + 8 = 32 + 255 * (40) + 8 = 10240
-//
-// Note that everything when serialized is padded to word length.
-//
-// type              = TX_START +  0*WORD_SIZE = 10240 +  0*8 = 10240
-// gasPrice          = TX_START +  1*WORD_SIZE = 10240 +  1*8 = 10248
-// gasLimit          = TX_START +  2*WORD_SIZE = 10240 +  2*8 = 10256
-// bytePrice         = TX_START +  3*WORD_SIZE = 10240 +  3*8 = 10264
-// maturity          = TX_START +  4*WORD_SIZE = 10240 +  4*8 = 10272
-// scriptLength      = TX_START +  5*WORD_SIZE = 10240 +  5*8 = 10280
-// scriptDataLength  = TX_START +  6*WORD_SIZE = 10240 +  6*8 = 10288
-// inputsCount       = TX_START +  7*WORD_SIZE = 10240 +  7*8 = 10296
-// outputsCount      = TX_START +  8*WORD_SIZE = 10240 +  8*8 = 10304
-// witnessesCount    = TX_START +  9*WORD_SIZE = 10240 +  9*8 = 10312
-// receiptsRoot      = TX_START + 10*WORD_SIZE = 10240 + 10*8 = 10320
-// SCRIPT_START      = TX_START + 11*WORD_SIZE = 10240 + 14*8 = 10352
-// SCRIPT_DATA_START = SCRIPT_START + SCRIPT_LENGTH
-
-const TX_TYPE_OFFSET = 10240;
-const TX_GAS_PRICE_OFFSET = 10248;
-const TX_GAS_LIMIT_OFFSET = 10256;
-const TX_BYTE_PRICE_OFFSET = 10264;
-const TX_MATURITY_OFFSET = 10272;
-const TX_SCRIPT_LENGTH_OFFSET = 10280;
-const TX_SCRIPT_DATA_LENGTH_OFFSET = 10288;
-const TX_INPUTS_COUNT_OFFSET = 10296;
-const TX_OUTPUTS_COUNT_OFFSET = 10304;
-const TX_WITNESSES_COUNT_OFFSET = 10312;
-const TX_RECEIPTS_ROOT_OFFSET = 10320;
-const TX_SCRIPT_START_OFFSET = 10352;
-const TX_ID_OFFSET = 0;
+const GTF_TYPE = 0x001;
+
+const GTF_SCRIPT_GAS_PRICE = 0x002;
+const GTF_SCRIPT_GAS_LIMIT = 0x003;
+const GTF_SCRIPT_MATURITY = 0x004;
+const GTF_SCRIPT_SCRIPT_LENGTH = 0x005;
+const GTF_SCRIPT_SCRIPT_DATA_LENGTH = 0x006;
+const GTF_SCRIPT_INPUTS_COUNT = 0x007;
+const GTF_SCRIPT_OUTPUTS_COUNT = 0x008;
+const GTF_SCRIPT_WITNESSES_COUNT = 0x009;
+const GTF_SCRIPT_RECEIPTS_ROOT = 0x00A;
+const GTF_SCRIPT_SCRIPT = 0x00B;
+const GTF_SCRIPT_SCRIPT_DATA = 0x00C;
```

Comment:
Now that I added these, I don't think I can even use them.

Reply:
I assume you're blocked by https://github.com/FuelLabs/sway/issues/810 for this right?